From 34b22c6fbab53db78a169bf13c6aefaebb3ec5b7 Mon Sep 17 00:00:00 2001
From: Chenyang Zhong <zhongcy95@gmail.com>
Date: Sun, 8 May 2022 04:07:54 -0400
Subject: [PATCH 4/4] DisplayPowerController: avoid updating settings when
 adjustment is NaN

E AndroidRuntime: *** FATAL EXCEPTION IN SYSTEM PROCESS: PowerManagerService
E AndroidRuntime: java.lang.IllegalArgumentException: Invalid value: NaN for setting: screen_auto_brightness_adj
E AndroidRuntime: 	at com.android.providers.settings.SettingsProvider.validateSystemSettingValue(SettingsProvider.java:1955)
E AndroidRuntime: 	at com.android.providers.settings.SettingsProvider.mutateSystemSetting(SettingsProvider.java:1923)
E AndroidRuntime: 	at com.android.providers.settings.SettingsProvider.insertSystemSetting(SettingsProvider.java:1843)
E AndroidRuntime: 	at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:462)
E AndroidRuntime: 	at android.content.ContentProvider.call(ContentProvider.java:2464)
E AndroidRuntime: 	at android.content.ContentProvider$Transport.call(ContentProvider.java:512)
E AndroidRuntime: 	at android.provider.Settings$NameValueCache.putStringForUser(Settings.java:2867)
E AndroidRuntime: 	at android.provider.Settings$System.putStringForUser(Settings.java:3574)
E AndroidRuntime: 	at android.provider.Settings$System.putStringForUser(Settings.java:3558)
E AndroidRuntime: 	at android.provider.Settings$System.putFloatForUser(Settings.java:3859)
E AndroidRuntime: 	at com.android.server.display.DisplayPowerController.putAutoBrightnessAdjustmentSetting(DisplayPowerController.java:2109)
E AndroidRuntime: 	at com.android.server.display.DisplayPowerController.updatePowerState(DisplayPowerController.java:1263)
E AndroidRuntime: 	at com.android.server.display.DisplayPowerController.access$700(DisplayPowerController.java:99)
E AndroidRuntime: 	at com.android.server.display.DisplayPowerController$DisplayControllerHandler.handleMessage(DisplayPowerController.java:2438)

Signed-off-by: Chenyang Zhong <zhongcy95@gmail.com>
Change-Id: Icb06aa8a2a72d6191be0b70b891bdac27a525fe3
---
 .../java/com/android/server/display/DisplayPowerController.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/display/DisplayPowerController.java b/services/core/java/com/android/server/display/DisplayPowerController.java
index 0d4834724e58..abc705684a7a 100644
--- a/services/core/java/com/android/server/display/DisplayPowerController.java
+++ b/services/core/java/com/android/server/display/DisplayPowerController.java
@@ -2128,7 +2128,7 @@ final class DisplayPowerController implements AutomaticBrightnessController.Call
     }
 
     private void putAutoBrightnessAdjustmentSetting(float adjustment) {
-        if (mDisplayId == Display.DEFAULT_DISPLAY) {
+        if (mDisplayId == Display.DEFAULT_DISPLAY && !Float.isNaN(adjustment)) {
             mAutoBrightnessAdjustment = adjustment;
             Settings.System.putFloatForUser(mContext.getContentResolver(),
                     Settings.System.SCREEN_AUTO_BRIGHTNESS_ADJ, adjustment,
-- 
2.32.0

